src/tensorlake/function_executor/proto/function_executor.proto

```diff
@@ -102,6 +102,23 @@ message Metrics {
     map<string, uint64> counters = 2;
 }
 
+enum FailureScope {
+    FAILURE_SCOPE_UNKNOWN = 0;
+    FAILURE_SCOPE_TASK = 1;        // The task failed, but can be retried.
+    FAILURE_SCOPE_INVOCATION = 2;  // The invocation cannot continue (e.g. invalid input).
+    FAILURE_SCOPE_GRAPH = 3;       // The graph is broken, requiring repair; block invocations.
```

Comment:
idk if this one would ever be useful tbh but okay.

Reply:
Removed